### PR TITLE
Fix duplicate arguments passed to dummy inputs in ONNX export

### PR DESCRIPTION
**Pull request by lewtun**
# What does this PR do?

This PR fixes: 

- a bug that was introduced in #15658 where a preprocessor and tokenizer were being passed together to the `generate_dummy_inputs()` function during the ONNX export.
- an oversight in the refactoring of the ONNX config for M2M-100

It also removes problematic TensorFlow integration tests, where the model implementation doesn't have parity with the PyTorch one (e.g. `camembert-base` is missing the causal LM head in TensorFlow). I'll address those issues in separate PRs as it involves touching the TensorFlow modeling files.

With these fixes, all slow ONNX tests now pass in all environments (only `torch`, only `tensorflow`, `torch` and `tensorflow`):

```bash
RUN_SLOW=1 python -m pytest tests/onnx/test_onnx_v2.py
```

cc @michaelbenayoun

### CODE REVIEW

File: src/transformers/onnx/convert.py [hunk 1]
```diff
@@ -100,11 +100,18 @@ def export_pytorch(
         `Tuple[List[str], List[str]]`: A tuple with an ordered list of the model's inputs, and the named inputs from
         the ONNX configuration.
     """
+    from ..tokenization_utils_base import PreTrainedTokenizerBase
```
Comment (LysandreJik) [inline src/transformers/onnx/convert.py:103]: I think this can be a top-level import

File: src/transformers/onnx/convert.py [hunk 2]
```diff
@@ -100,11 +100,18 @@ def export_pytorch(
...
+
+    if isinstance(preprocessor, PreTrainedTokenizerBase) and tokenizer is not None:
+        raise ValueError("You cannot provide both a tokenizer and a preprocessor to export the model.")
     if tokenizer is not None:
         warnings.warn(
             "The `tokenizer` argument is deprecated and will be removed in version 5 of Transformers. Use `preprocessor` instead.",
             FutureWarning,
         )
+        logger.warning("Overwriting the `preprocessor` argument with `tokenizer` to generate dummmy inputs.")
```
Comment (LysandreJik) [inline src/transformers/onnx/convert.py:112]: Maybe this can be an `info` as it's more additional information and not really an error

Comment (LysandreJik) [reply src/transformers/onnx/convert.py:112]: (warnings get displayed by default, info is displayed when users ask to have more info)

File: src/transformers/onnx/convert.py [hunk 3]
```diff
@@ -213,11 +218,17 @@ def export_tensorflow(
...
+    from ..tokenization_utils_base import PreTrainedTokenizerBase
+
+    if isinstance(preprocessor, PreTrainedTokenizerBase) and tokenizer is not None:
+        raise ValueError("You cannot provide both a tokenizer and preprocessor to export the model.")
     if tokenizer is not None:
         warnings.warn(
             "The `tokenizer` argument is deprecated and will be removed in version 5 of Transformers. Use `preprocessor` instead.",
             FutureWarning,
         )
+        logger.warning("Overwriting the `preprocessor` argument with `tokenizer` to generate dummmy inputs.")
```
Comment (LysandreJik) [inline src/transformers/onnx/convert.py:230]: Same comment about logging level

**Review by LysandreJik — APPROVED** [review]
Looks good! Left two comments that should be applied 4 times each :)

**Review by sgugger — APPROVED** [review]
Good for me with Lysandre's comments! Thanks for working on this!

**Pull request merged**